### ISSUE TITLE
Can you calculate a distance matrix between point patterns without calculating all vertex distance pairs?

### ISSUE THREAD

**Issue by iboates** [pysal/spaghetti#725]
I am struggling with some performance issues when calling [`Network. allneighbordistances`](https://github.com/pysal/spaghetti/blob/7dfb5e36c141086aa080c596fca648bbbe82af86/spaghetti/network.py#L1658) between my source and destination point patterns.

When I looked into the code, it seems to me that the first thing this method does is calculate all the vertex distance pairs in the entire network with [`Network.full_distance_matrix`](https://github.com/pysal/spaghetti/blob/7dfb5e36c141086aa080c596fca648bbbe82af86/spaghetti/network.py#L1570)

Is there a way to only compute the network distances between the source and destination point patterns without computing all vertex pair distances? Maybe I am not getting it but it seems wasteful to do all vertex pairs first when I already know the distances of every line feature from the geodataframe from which I initialized the network.

For reference I am using spaghetti in tandem with spopt to optimize the placement of facilities, mostly by following (but modifying) this tutorial: https://pysal.org/spopt/notebooks/mclp.html

**Comment by jGaboardi** [comment]
No, this is not currently possible due to the historic structure of `pysal.network` (`spaghetti`'s predecessor) and the utilization of the [`djikstra `](https://github.com/pysal/spaghetti/blob/7dfb5e36c141086aa080c596fca648bbbe82af86/spaghetti/util.py#L155) algorithm). There has been discussion[^1][^2] for overhauling & modernizing `spaghetti` for some years now, but there has been little action due to time constraints. And as usual, since this is an OS project maintained in our free time contributions are always welcome!



[^1]: https://github.com/pysal/spaghetti/discussions/195
[^2]: https://github.com/pysal/spaghetti/discussions/403

**Comment by jGaboardi** [comment]
* `spaghetti` is reaching End of Life – #780 
* no new functionality or bug fixes are planned
